### PR TITLE
Update polkadot deps to latest version v0.9.43

### PR DESCRIPTION
**Pull request by salman01zp**
## Summary of changes

Update to v0.9.43

- https://github.com/paritytech/substrate/pull/13592
Moves code from sc-network-common  to sc-network

- https://github.com/paritytech/substrate/pull/13425
Moves sc-finality-grandpa to sc-consensus-grandpa

- https://github.com/paritytech/substrate/pull/12828
extracts the syncing protocol from sc-network

Polkadot release notes: https://github.com/paritytech/polkadot/releases
## NOTE
There are some breaking changes around syncing protocol and Grandpa. CC @1xstj 

---
**Reference issue (if applicable)**

Refers:  https://github.com/webb-tools/tangle/issues/201

### CODE REVIEW

File: dkg-gadget/src/gossip_engine/network.rs [hunk 1]
```diff
@@ -390,23 +388,6 @@ impl<B: Block + 'static> GossipHandler<B> {
...
-					self.logger.error(format!("Add reserved peer failed: {err}"));
-				}
-			},
-			Event::SyncDisconnected { remote } => {
-				self.service.remove_peers_from_reserved_set(
-					self.protocol_name.clone(),
-					iter::once(remote).collect(),
-				);
-			},
-
```
Comment (shekohex) [inline dkg-gadget/src/gossip_engine/network.rs:409]: Where are these events handlers gone? what are the alternatives?

File: pallets/dkg-metadata/src/lib.rs [hunk 2]
```diff
@@ -409,11 +408,11 @@ pub mod pallet {
 			if Self::should_refresh(n) && !Self::refresh_in_progress() {
 				if let Some(pub_key) = Self::next_dkg_public_key() {
 					Self::do_refresh(pub_key.1.into());
-					return Weight::from_ref_time(1_u64)
+					return Weight::from_parts(1_u64, 1024)
```
Comment (shekohex) [inline pallets/dkg-metadata/src/lib.rs:411]: what is the `1024` value here?

File: standalone/node/src/service.rs [hunk 3]
```diff
@@ -293,15 +268,14 @@ pub fn new_full(
 		);
 
 		let relayer_params = webb_relayer_gadget::WebbRelayerParams {
-			local_keystore: keystore_container.local_keystore(),
+			local_keystore: Some(keystore_container.local_keystore()),
```
Comment (shekohex) [inline standalone/node/src/service.rs:271]: if it is always there, there is no need to wrap it into Option::Some, you can change the `WebbRelayerParams` to take that value directly.

**Review by shekohex — CHANGES_REQUESTED** [review]

File: pallets/dkg-metadata/src/lib.rs (hunk 2 above)
Comment (salman01zp) [reply pallets/dkg-metadata/src/lib.rs:411]: Measured:  1
Estimated: 1024

File: dkg-gadget/src/gossip_engine/network.rs (hunk 1 above)
Comment (salman01zp) [reply dkg-gadget/src/gossip_engine/network.rs]: they moved it to separate SyncEvents, updated.

**Review by 1xstj — APPROVED** [review]

**Pull request merged**